harmony/harmonydb/sql/20240930-pdp.sql

```diff
@@ -68,20 +68,31 @@ CREATE TABLE pdp_piece_mh_to_commp (
 CREATE INDEX pdp_piecerefs_piece_cid_idx ON pdp_piecerefs(piece_cid);
 
 -- PDP proofsets we maintain
+
 CREATE TABLE pdp_proof_sets (
     id BIGINT PRIMARY KEY, -- on-chain proofset id
 
     -- updated when a challenge is requested (either by first proofset add or by invokes of nextProvingPeriod)
     -- initially NULL on fresh proofsets.
     prev_challenge_request_epoch BIGINT,
 
-    -- task invoking nextProvingPeriod, the task should be spawned any time prev_challenge_request_epoch+provingPeriod is in the past
+    -- task invoking nextProvingPeriod, the task should be spawned any time prove_at_epoch+challenge_window is in the past
     challenge_request_task_id BIGINT REFERENCES harmony_task(id) ON DELETE SET NULL,
 
     -- first proofset add or nextProvingPeriod message hash, when the message lands prove_task_id will be spawned and
     -- this value will be set to NULL
     challenge_request_msg_hash TEXT,
 
+    -- the listener is assumed to be a PDP Service with proving window informational methods
+    proofset_listener TEXT NOT NULL,
```

Comment:
Any way we can just query this from the Verifier contract? Less off-chain state is generally always desirable, especially if it has any remote possibility of being incorrect here.

Reply:
Yes this shouldn't be a problem.